--- net/minecraft/net/minecraft/block/BlockNote.java
+++ net/minecraft/net/minecraft/block/BlockNote.java
@@ -50,9 +50,7 @@
     }
 
     public boolean onBlockActivated(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
-        if (worldIn.isRemote) {
-            return true;
-        } else {
+        if (!worldIn.isRemote) {
             TileEntity tileentity = worldIn.getTileEntity(pos);
 
             if (tileentity instanceof TileEntityNote) {
@@ -61,9 +59,12 @@
                 tileentitynote.triggerNote(worldIn, pos);
                 playerIn.addStat(StatList.NOTEBLOCK_TUNED);
             }
-
-            return true;
         }
+        return true;
+    }
+    
+    public boolean canActivateBlock(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
+        return true;
     }
 
     public void onBlockClicked(World worldIn, BlockPos pos, EntityPlayer playerIn) {
