--- net/minecraft/net/minecraft/block/Block.java
+++ net/minecraft/net/minecraft/block/Block.java
@@ -637,6 +637,10 @@
     public boolean onBlockActivated(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
         return false;
     }
+    
+    public boolean canActivateBlock(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
+        return false;
+    }
 
     /**
      * Triggered whenever an entity collides with this block (enters into the block)
