--- net/minecraft/block/BlockRedstoneRepeater.java
+++ net/minecraft/block/BlockRedstoneRepeater.java
@@ -68,6 +68,10 @@
             return true;
         }
     }
+    
+    public boolean canActivateBlock(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
+        return playerIn.capabilities.allowEdit;
+    }
 
     protected int getDelay(IBlockState state) {
         return ((Integer)state.getValue(DELAY)).intValue() * 2;
