--- net/minecraft/block/BlockDispenser.java
+++ net/minecraft/block/BlockDispenser.java
@@ -85,9 +85,7 @@
     }
 
     public boolean onBlockActivated(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
-        if (worldIn.isRemote) {
-            return true;
-        } else {
+        if (!worldIn.isRemote) {
             TileEntity tileentity = worldIn.getTileEntity(pos);
 
             if (tileentity instanceof TileEntityDispenser) {
@@ -99,9 +97,12 @@
                     playerIn.addStat(StatList.DISPENSER_INSPECTED);
                 }
             }
-
-            return true;
         }
+        return true;
+    }
+    
+    public boolean canActivateBlock(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
+        return true;
     }
 
     protected void dispense(World worldIn, BlockPos pos) {
