--- net/minecraft/net/minecraft/block/BlockJukebox.java
+++ net/minecraft/net/minecraft/block/BlockJukebox.java
@@ -47,6 +47,10 @@
             return false;
         }
     }
+    
+    public boolean canActivateBlock(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
+        return ((Boolean)state.getValue(HAS_RECORD)).booleanValue();
+    }
 
     public void insertRecord(World worldIn, BlockPos pos, IBlockState state, ItemStack recordStack) {
         TileEntity tileentity = worldIn.getTileEntity(pos);
