--- net/minecraft/block/BlockBed.java
+++ net/minecraft/block/BlockBed.java
@@ -116,6 +116,10 @@
             }
         }
     }
+    
+    public boolean canActivateBlock(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
+        return true;
+    }
 
     @Nullable
     private EntityPlayer getPlayerInBed(World worldIn, BlockPos pos) {
