--- net/minecraft/block/BlockRedstoneComparator.java
+++ net/minecraft/block/BlockRedstoneComparator.java
@@ -158,6 +158,10 @@
             return true;
         }
     }
+    
+    public boolean canActivateBlock(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
+        return playerIn.capabilities.allowEdit;
+    }
 
     protected void updateState(World worldIn, BlockPos pos, IBlockState state) {
         if (!worldIn.isBlockTickPending(pos, this)) {
