--- net/minecraft/net/minecraft/block/BlockDoor.java
+++ net/minecraft/net/minecraft/block/BlockDoor.java
@@ -134,6 +134,17 @@
             }
         }
     }
+    
+    public boolean canActivateBlock(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
+        if (this.blockMaterial == Material.IRON) {
+            return false;
+        } else {
+            BlockPos blockpos = state.getValue(HALF) == BlockDoor.EnumDoorHalf.LOWER ? pos : pos.down();
+            IBlockState iblockstate = pos.equals(blockpos) ? state : worldIn.getBlockState(blockpos);
+
+            return iblockstate.getBlock() == this;
+        }
+    }
 
     public void toggleDoor(World worldIn, BlockPos pos, boolean open) {
         IBlockState iblockstate = worldIn.getBlockState(pos);
