--- net/minecraft/block/BlockStructure.java
+++ net/minecraft/block/BlockStructure.java
@@ -37,6 +37,10 @@
         TileEntity tileentity = worldIn.getTileEntity(pos);
         return tileentity instanceof TileEntityStructure ? ((TileEntityStructure)tileentity).usedBy(playerIn) : false;
     }
+    
+    public boolean canActivateBlock(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
+        return worldIn.getTileEntity(pos) instanceof TileEntityStructure ? true : false;
+    }
 
     /**
      * Called by ItemBlocks after a block is set in the world, to allow post-place logic
