--- net/minecraft/net/minecraft/block/BlockCauldron.java
+++ net/minecraft/net/minecraft/block/BlockCauldron.java
@@ -198,6 +198,10 @@
             }
         }
     }
+    
+    public boolean canActivateBlock(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
+        return true;
+    }
 
     public void setWaterLevel(World worldIn, BlockPos pos, IBlockState state, int level) {
         worldIn.setBlockState(pos, state.withProperty(LEVEL, Integer.valueOf(MathHelper.clamp(level, 0, 3))), 2);
