--- net/minecraft/net/minecraft/block/BlockPistonMoving.java
+++ net/minecraft/net/minecraft/block/BlockPistonMoving.java
@@ -106,6 +106,10 @@
             return false;
         }
     }
+    
+    public boolean canActivateBlock(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
+        return !worldIn.isRemote && worldIn.getTileEntity(pos) == null;
+    }
 
     /**
      * Get the Item that this Block should drop when harvested.
