--- net/minecraft/net/minecraft/block/BlockDragonEgg.java
+++ net/minecraft/net/minecraft/block/BlockDragonEgg.java
@@ -71,6 +71,10 @@
         this.teleport(worldIn, pos);
         return true;
     }
+    
+    public boolean canActivateBlock(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing heldItem, float side, float hitX, float hitY) {
+        return true;
+    }
 
     public void onBlockClicked(World worldIn, BlockPos pos, EntityPlayer playerIn) {
         this.teleport(worldIn, pos);
